config/config_test.go

```diff
@@ -1 +1,476 @@
 package config
+
+import (
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/apis/extensions"
+	"k8s.io/kubernetes/pkg/client/unversioned/testclient"
+	"k8s.io/kubernetes/pkg/util/intstr"
+)
+
+func testDir(t *testing.T) (string, func()) {
+	dir, err := ioutil.TempDir("", "haproxy_config")
+	if err != nil {
+		t.Fatal("failed to build dir")
+	}
+	return dir, func() { defer os.RemoveAll(dir) }
+}
+
+func TestNewConfig(t *testing.T) {
+	dir, cleanup := testDir(t)
+	defer cleanup()
+
+	c := NewConfig(&testclient.FakeIngress{}, dir+"/file", "local.cluster")
+	if c == nil {
```

Comment:
I think `NewConfig` should return an error if there is a possibility of it returning a nil pointer.